Convert capa/display.coffee to JS.

cahrens opened the pull request:
Reviewers:
- [x] @staubina 
- [x] @muhammad-ammar 

This file had 628 eslint violations (not including indenting issues)!

To see the code changes in the JS from the CoffeeScript-generated code (not including indenting changes), see the fourth commit (580bcaf9a2dbedc2b042cccbe7841f73e5fb616f).

Sandbox: https://capacoffee.sandbox.edx.org

cahrens commented:
FYI @muhammad-ammar @staubina I will be addressing your feedback to #13704 on this branch. I will let you know when this PR is ready for re-review.

cahrens commented:
@staubina and @muhammad-ammar this is ready for re-review, and the sandbox is updated.

cahrens commented:
jenkins run bokchoy

cahrens commented:
jenkins run lettuce

cahrens commented:
jenkins run bokchoy

staubina commented:
👍 
- [ ] Reviewed the changes.
- [ ] Tested on the Sandbox for any issues with the Capa updates done recently.

cahrens commented:
@muzaffaryousaf Please review this PR tomorrow. The last commit on the PR are the changes I made in response to @muhammad-ammar's review in #13704. Thank you!

muzaffaryousaf commented on common/lib/xmodule/xmodule/js/src/capa/display.js:
```diff
@@ -0,0 +1,1319 @@
+        Problem.prototype.queueing = function(focusCallback) {
+            var that = this;
+            this.queued_items = this.$('.xqueue');
+            this.num_queued_items = this.queued_items.length;
```
how about JS naming convention in this file, do we need to address such kind of things ?

cahrens replied:
I only changed the things causing eslint violations. For some reason, this one wasn't flagged.

muzaffaryousaf commented on common/lib/xmodule/xmodule/js/src/capa/display.js:
```diff
@@ -0,0 +1,1319 @@
+             *     'enable' is a boolean to determine enabling/disabling of submit button.
+             *     'changeText' is a boolean to determine if there is need to change the
+             *         text of submit button as well.
+             */
```
Don't we add docs string before the function name ?

muzaffaryousaf commented:
I've verified changes made in response to @muhammad-ammar 's feedback. Also, i've asked couple of questions. Otherwise this LGTM 👍

muzaffaryousaf commented on common/lib/xmodule/xmodule/js/src/capa/display.js:
```diff
@@ -0,0 +1,1319 @@
+            // Render 'x/y point(s)' if student has attempted question
+            if (status !== 'none' && (detail !== null && detail !== undefined) && (jQuery.type(detail) === 'string') &&
+                detail.indexOf('/') > 0) {
+                a = detail.split('/');
```
After deeper look, there are many variable like `a` that are being used in this file, do you think we should change the names as part of this work ?

muzaffaryousaf commented on common/lib/xmodule/xmodule/js/src/capa/display.js:
```diff
@@ -0,0 +1,1319 @@
+            }
+            this.enableSubmitButton(false);
+            if (!window.FormData) {
+                alert(gettext('Submission aborted! Sorry, your browser does not support file uploads. If you can, please use Chrome or Safari which have been verified to support file uploads.')); // eslint-disable-line max-len, no-alert
```
`gettext` on new separate line ?

muzaffaryousaf replied:
wrong call.

muzaffaryousaf commented on common/lib/xmodule/xmodule/js/src/capa/display.js:
```diff
@@ -0,0 +1,1319 @@
+                }
+            });
+            if (fileNotSelected) {
+                errors.push(gettext('You did not select any files to submit.'));
```
gettext on new line ?

muzaffaryousaf commented on common/lib/xmodule/xmodule/js/src/capa/display.js:
```diff
@@ -0,0 +1,1323 @@
+                    return preprocessor.fn;
+                }
+            },
+            javascriptinput: function(element) {
```
`javascriptinput` --> `javascriptInput`

cahrens replied:
@muzaffaryousaf I'm going to stick to only changing things that cause eslint warnings. There are obviously still many things that could be improved about this code!

muzaffaryousaf commented:
@cahrens I'm done reviewing this, my only concern is variable/functions 's naming convention.

muzaffaryousaf commented on common/lib/xmodule/xmodule/js/src/capa/display.js:
```diff
@@ -0,0 +1,1319 @@
+                    if (ariaLabel) {
+                        // Translators: This is only translated to allow for reordering of label and associated status.;
+                        template = gettext('{label}: {status}');
+                        labeledStatus.push(edx.StringUtils.interpolate(template, {
```
My understanding is: we usually keep gettext and interpolate on different lines. Not sure if that's only limited to underscore.

muzaffaryousaf replied:
> Due to a bug in the underlying underscore extraction library, when StringUtils.interpolate and gettext are on the same line, the library will not work properly. In such cases, the library will extract the word gettext rather than the actual string that needs to be extracted. Make sure to separate StringUtils.interpolate and gettext into two different lines, as shown in the example above.

cahrens replied:
Thanks for that information-- I had forgotten that. But in this example, they are on separate lines.

cahrens replied:
Unless you think that the variable "template" will be removed and the content in-lined before the extraction?

cahrens replied:
There are couple of examples like this in the file. I'll change them just to be safe.

cahrens commented:
Thanks for the review @muzaffaryousaf. As I mentioned, for the purposes of this conversion I prefer to leave variable and function names as-is unless they are flagged by eslint.

I have addressed your other feedback and squashed those changes into the final "eslint" commit. I will merge after getting a green build.

Thanks for stepping in and helping out with the review!

cahrens commented:
jenkins run bokchoy
